packages/financial-templates-lib/src/logger/Transports.ts

```diff
@@ -80,14 +80,20 @@ export function createTransports(transportsConfig: TransportsConfig = {}): Trans
         )
       );
     }
+    // typescript is not properly typing this as a parsed json string should be "unknown",
+    // but we are relying on this quirk throughout this file. The PD v2 transport class will check
+    // that the config is valid or throw an error to prevent errors while running.
+    const pagerDutyV2Config =
+      transportsConfig.pagerDutyV2Config ?? JSON.parse(process.env.PAGER_DUTY_V2_CONFIG || "null");
     // If there is a Pagerduty V2 API key then add the pagerduty winston transport.
-    if (transportsConfig.pagerDutyV2Config || process.env.PAGER_DUTY_V2_CONFIG) {
-      transports.push(
-        new PagerDutyV2Transport(
-          { level: "warn" },
-          transportsConfig.pagerDutyV2Config ?? JSON.parse(process.env.PAGER_DUTY_V2_CONFIG || "null")
-        )
-      );
+    if (pagerDutyV2Config) {
+      try {
+        transports.push(new PagerDutyV2Transport({ level: "warn" }, pagerDutyV2Config));
+      } catch (err) {
+        if (err instanceof Error) {
+          console.warn("Pagerduty V2 Config not added due to error: ", err.message);
```

Comment:
is this a good idea? it might be better to throw the error.

console.warn wont propagate in serverless environments necessarily: we totally rely on the winston logs there